shell.nix

```diff
@@ -0,0 +1,29 @@
+let
```

Comment:
Can you add yourself to `.github/CODEOWNERS` for ownership of this file?

Reply:
Sorry, but how is this related to Promscale. I feel its very specific to your system?

Reply:
It is true that I use this on my laptop, but it could be used by others who want to use nix, or even in CI. The idea behind nix is to have a deterministic way to install tools and dependencies on a per-project basis. I've only added a couple of "random" dependencies for now, but I plan on adding go for instance, so that the version of go used for Promscale is independent from my system's go version.